src/backend/evalu8.c

```diff
@@ -270,6 +270,41 @@ int boolres(elem *e)
                     break;
                 }
 
+                case TYschar32:
+                case TYuchar32:
+                case TYshort16:
+                case TYushort16:
+                case TYlong8:
+                case TYulong8:
+                case TYllong4:
+                case TYullong4:
+                    b = 0;
+                    for (size_t i = 0; i < 8; i++)
+                        b |= e->EV.Vulong8[i] != 0;
```

Comment:
1. It's not doing movs.
2. There's a 32 bit dmd build, too.
3. It doesn't matter. This piece of code is nowhere near a hot spot.